### PR TITLE
Refactor cartPresenter to reflect the real business need

### PR DESCRIPTION
**Pull request by arouiadib**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | This PR just rewrites a private method signature, see details below
| Type?             |  refacto 
| Category?         | FO 
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | N/A
| How to test?      | N/A
| Possible impacts? | N/A

1. private method `cartVoucherHasFreeShipping()` wasn't used so I removed it.
2. However, a naming that reflects the business more is `cartVoucherHasFreeShippingOnly()`, so I created this private method.
3. I refactored an if statement with trailing conditions to a simple call to the new method.
Thanks for your time

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24940)

### CODE REVIEW

File: src/Adapter/Presenter/Cart/CartPresenter.php [hunk 1]
```diff
@@ -656,6 +644,18 @@ private function cartVoucherHasGiftProductReduction($cartVoucher)
         return !empty($cartVoucher['gift_product']);
     }
 
+    /**
+     * @param array $cartVoucher
+     *
+     * @return bool
+     */
+    private function cartVoucherHasFreeShippingOnly($cartVoucher)
```
Comment (Progi1984) [inline src/Adapter/Presenter/Cart/CartPresenter.php]:
```suggestion
    private function cartVoucherHasFreeShippingOnly(array $cartVoucher): bool
```

**Review by Progi1984 — COMMENTED** [review]
A small feedback

**Comment by arouiadib** [comment]
@Progi1984 there is a couple of other similar methods (with array argument and boolean return) that doesn't have your remark. I think it will create no issue if I add these changes for them as well,  in this PR? wdyt?

**Review by Progi1984 — DISMISSED** [review]

**Comment by Progi1984** [comment]
@arouiadib They are private ?

**Comment by arouiadib** [comment]
@Progi1984 , yes

**Comment by matks** [comment]
> private method cartVoucherHasFreeShipping() wasn't used so I removed it.
However, a naming that reflects the business more is cartVoucherHasFreeShippingOnly(), so I created this private method.
I refactored an if statement with trailing conditions to a simple call to the new method.
Thanks for your time

Thank you for the detailed description 👍

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Review by matks — APPROVED** [review]
As this is only code being moved and functions being merged together into a private method I don't think QA is needed on this PR 🤔

**Comment by Progi1984** [comment]
Thanks @arouiadib

**Pull request merged**